src/components/ListAgents/ListAgentsWithVCs.tsx

```diff
@@ -110,6 +112,16 @@ export default function ListAgentsWithVCs({
     }
   }, [accessEndpoint, endpointConfiguration, session, isValidAccessGrant]);
 
+  useEffect(() => {
+    if (typeof onAgentUpdate === "function") {
+      let agentsCount = 0;
+      if (agents !== undefined) {
+        agentsCount = agents.length;
+      }
+      onAgentUpdate(agentsCount);
```

Comment:
```suggestion
      onAgentUpdate((agents ?? []).length);
```
I don't have a strong opinion on which one we should use, feel free to keep the one you find easier to read and understand.